CDP4CrossViewEditor/Generator/CrossviewSheetPMeanUtilities.cs

```diff
@@ -0,0 +1,159 @@
+    using System.Globalization;
+    using System.Linq;
+
+    using CDP4Common.EngineeringModelData;
+
+    /// <summary>
+    /// Helper class that is required for computing value of P_mean parameter
+    /// </summary>
+    internal static class CrossviewSheetPMeanUtilities
+    {
+        /// <summary>
+        /// Hardcoded redundancy.scheme
+        /// NOTE: not sure about this value
+        /// </summary>
+        private const string RedundancySchemeHot = "Active";
+
+        /// <summary>
+        /// Hardcoded redundancy.scheme
+        /// NOTE: not sure about this value
+        /// </summary>
+        private const string RedundancySchemeCold = "Passive";
+
+        /// <summary>
+        /// Hardcoded redundancy.type
+        /// </summary>
+        private const string RedundancyTypeInternal = "Internal";
+
+        /// <summary>
+        /// Hardcoded redundancy.type
+        /// </summary>
+        private const string RedundancyTypeExternal = "External";
+
+        /// <summary>
+        /// Check if P_mean parameter calculation is possible
+        /// </summary>
+        /// <param name="parameters"></param>
+        /// <returns>True if all power related parameters are present, False otherwise</returns>
+        public static bool IsCalculationPossible(List<ParameterOrOverrideBase> parameters)
+        {
+            // Check if all power required parameters are present
+            if (CrossviewSheetConstants.PowerParameters.Except(new[] { "P_peak" }).Except(parameters.Select(p => p.ParameterType.ShortName)).Any())
+            {
+                return false;
+            }
+
+            var pDutyCycle = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_duty_cyc", StringComparison.InvariantCultureIgnoreCase));
+            var pMean = parameters.Where(p => p != null)
+                .FirstOrDefault(p => p.ParameterType.ShortName.Equals("P_mean", StringComparison.InvariantCultureIgnoreCase));
+
+            if (pDutyCycle == null || pMean == null)
+            {
+                return false;
+            }
+
+            // Check if P_duty_cyc && P_mean have the same option dependency
+
+            if (pDutyCycle.IsOptionDependent != pMean.IsOptionDependent)
+            {
+                return false;
+            }
+
+            if (pDutyCycle.IsOptionDependent &&
+                pMean.IsOptionDependent &&
+                pDutyCycle.ValueSets.Select(vs => vs.ActualOption).Except(pMean.ValueSets.Select(vs => vs.ActualOption)).Any())
+            {
+                return true;
+            }
+
+            // Check if P_duty_cyc && P_mean have the same state dependency, both should be
+
+            if (pDutyCycle.StateDependence == null || pMean.StateDependence == null)
+            {
+                return false;
+            }
+
+            if (pDutyCycle.StateDependence.ActualState.Count == 0 || pMean.StateDependence.ActualState.Count == 0)
```

Comment:
This check is not strong enough, there's no need to compare the state counts, it should be enough that both have the same StateDependence (of type ActualFiniteStateList): `pDutyCycle.StateDependence == pMean.StateDependence`.